faq/osgi/README.md

```diff
@@ -0,0 +1,205 @@
+    </dependency>
+
+Make sure that your project's manifest is importing the right version
+of Guava's packages, e.g. for 19.0:
+
+    Import-Package: com.google.common.base;version="[19.0,20)"
+
+
+### How to enable compression?
+
+First, read our [manual page on compression](../../manual/compression/)
+to understand how to enable compression for the Java driver.
+            
+OSGi projects can use both Snappy or LZ4 compression algorithms. 
+
+For Snappy, include the following Maven dependency:
+
+    <dependency>
+        <groupId>org.xerial.snappy</groupId>
+        <artifactId>snappy-java</artifactId>
+        <version>1.1.2.6</version>
+    </dependency>
+
+For LZ4, include the following Maven dependency:
+
+    <dependency>
+        <groupId>net.jpountz.lz4</groupId>
+        <artifactId>lz4</artifactId>
+        <version>1.3.0</version>
+    </dependency>
+
+**IMPORTANT**: versions of LZ4 library below 1.3.0 cannot be used
+because they are _not_ valid OSGi bundles.
+
+Because compression libraries are _optional runtime dependencies_, 
+most manifest generation tools (such as [BND] and the [Maven bundle plugin]) 
+will not reference these libraries in your project's manifest.
+This is correct, but could be a problem for some OSGi provisioning tools,
+and notably for [Tycho], because it does not consider such
+dependencies when computing the target platform.
+
+If you are facing provisioning issues related to compression libraries,
+you might need to either add them explicitly to your OSGi runtime,
+or explicitly reference them in your project's manifest.
+With the [Maven bundle plugin], this second option can be achieved with the following 
+[BND] `Import-Package` instruction (the example below is for
+LZ4, but the same applies to Snappy as well):
+
+    <Import-Package>net.jpountz.lz4,*</Import-Package>
+
+With Tycho, another option is to explicitly declare an "extra requirement"
+to the compression library in the target platform definition:
+
+    <plugin>
+        <groupId>org.eclipse.tycho</groupId>
+        <artifactId>target-platform-configuration</artifactId>
+        <version>0.25.0</version>
+        <configuration>
+            <dependency-resolution>
+                <extraRequirements>
+                    <requirement>
+                        <id>lz4-java</id>
+                        <versionRange>1.3.0</versionRange>
+                        <type>eclipse-plugin</type>
+                    </requirement>
+                </extraRequirements>
+            </dependency-resolution>
+            ...
+        </configuration>
+    </plugin>
+    
+Note that the requirement id is its bundle symbolic name,
+_not_ its Maven artifact id.
+
+    
+### How to use the driver shaded jar?
+
+The driver [shaded jar](../../manual/shaded_jar/) can be used 
+in any OSGi application, although the same limitations explained in
+the manual apply.
+
+Be sure _not_ to include any non-shaded Netty bundle in your project.
```

Comment:
Shouldn't this still work? The driver will just not use it, but other libraries that depend on netty should be able to use that version right?